tests/ui/UnreadIndicatorsTest.js

```diff
@@ -0,0 +1,395 @@
+/* eslint-disable @lwc/lwc/no-async-await */
+import React from 'react';
+import Onyx from 'react-native-onyx';
+import {Linking, AppState} from 'react-native';
+import {fireEvent, render, act} from '@testing-library/react-native';
+import lodashGet from 'lodash/get';
+import moment from 'moment';
+import App from '../../src/App';
+import CONST from '../../src/CONST';
+import ONYXKEYS from '../../src/ONYXKEYS';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+import * as TestHelper from '../utils/TestHelper';
+import appSetup from '../../src/setup';
+import fontWeightBold from '../../src/styles/fontWeight/bold';
+import * as AppActions from '../../src/libs/actions/App';
+import ReportHeaderViewBackButton from '../../src/pages/home/ReportHeaderViewBackButton';
+import ReportActionsView from '../../src/pages/home/report/ReportActionsView';
+import * as NumberUtils from '../../src/libs/NumberUtils';
+import LocalNotification from '../../src/libs/Notification/LocalNotification';
+import * as Report from '../../src/libs/actions/Report';
+
+beforeAll(() => {
+    global.fetch = TestHelper.getGlobalFetchMock();
+
+    // We need a bit more time for this test in some places
+    jest.setTimeout(30000);
+    Linking.setInitialURL('https://new.expensify.com/r/1');
+    appSetup();
+});
+
+/**
+ * @param {RenderAPI} renderedApp
+ */
+function scrollUpToRevealNewMessagesBadge(renderedApp) {
+    fireEvent.scroll(renderedApp.getByTestId('report-actions-list'), {
+        nativeEvent: {
+            contentOffset: {
+                y: 250,
+            },
+            contentSize: {
+                // Dimensions of the scrollable content
+                height: 500,
+                width: 100,
+            },
+            layoutMeasurement: {
+                // Dimensions of the device
+                height: 700,
+                width: 300,
+            },
+        },
+    });
+
+    // We advance the timer since we must wait for the animation to end
+    // and the new style to be reflected
+    jest.advanceTimersByTime(100);
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Boolean}
+ */
+function isNewMessagesBadgeVisible(renderedApp) {
+    const badge = renderedApp.getByTestId('new-messages-badge');
+    return badge.props.style.transform[0].translateY === 10;
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Promise}
+ */
+async function navigateToSidebar(renderedApp) {
+    const reportHeader = renderedApp.getByTestId('report-header');
+    const reportHeaderBackButton = reportHeader.findByType(ReportHeaderViewBackButton);
```

Comment:
A lot of the reading I did gave me the impression that querying by type was a bad thing and it was deprecated. Did you ever come across any of that? Also, why use `findBy*` here instead of `getBy`?

Reply:
I am still working on my expertise with this library, but looks like you're correct that `getByType()` has been deprecated. AFAICT `findBy` queries are not. I didn't come across anything other than that the [usage is discouraged](https://callstack.github.io/react-native-testing-library/docs/migration-v2/#removed-functions).

Unsure if there's any other implied issue with using `findBy*` or querying by type - let me know if there's something specific you read! 

https://callstack.github.io/react-native-testing-library/docs/api-queries/#findby

Then again, maybe this ties into your concern about consistency about the kinds of selectors we should use. IIRC I ended up going with "type" here because of some issues triggering the "press" event, but maybe I can clean this up with the `accessibilityHint` trick. Gonna give that a try.

Reply:
I got rid of this usage. There's one other spot where I left in the access by type, but I think I can get rid of it as well. Since the thing I want to know is whether the "drawer is open" I _think_ we could give an `accessibilityElementsHidden` property to the sidebar and check that.

Reply:
It worked!

Reply:
Yeah, it's weird that `findByType` or `queryByType` wouldn't also be unsafe. Very strange. You found the same stuff that I did, but I don't think it registered to me that find and query by type were OK.

Great that the accessibility stuff worked out!